File: uhabits-android/src/androidTest/java/org/isoron/uhabits/activities/habits/list/views/HintViewTest.java
/*
 * Copyright (C) 2016-2021 Álinson Santos Xavier <git@axavier.org>
 *
 * This file is part of Loop Habit Tracker.
 *
 * Loop Habit Tracker is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by the
 * Free Software Foundation, either version 3 of the License, or (at your
 * option) any later version.
 *
 * Loop Habit Tracker is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package org.isoron.uhabits.activities.habits.list.views;

import androidx.test.filters.*;
import androidx.test.runner.*;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.isoron.uhabits.*;
import org.isoron.uhabits.core.ui.screens.habits.list.*;
import org.junit.*;
import org.junit.runner.*;

import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.MatcherAssert.*;
import static org.mockito.Mockito.*;

@RunWith(AndroidJUnit4.class)
@MediumTest
public class HintViewTest extends BaseViewTest
{
    public static final String PATH = "habits/list/HintView/";

    private HintView view;

    private HintList list;

    @Before
    @Override
    public void setUp()
    {
        super.setUp();

        list = mock(HintList.class);
        view = new HintView(targetContext, list);
        measureView(view, 400, 200);

        String text =
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";

        when(list.shouldShow()).thenReturn(true);
        when(list.pop()).thenReturn(text);

        view.showNext();
        skipAnimation(view);
    }

    @Test
    public void testRender() throws Exception
    {
        assertRenders(view, PATH + "render.png");
    }

    @Test
    public void testClick() throws Exception
    {
        assertThat(view.getAlpha(), equalTo(1f));
        view.performClick();
        skipAnimation(view);
        assertThat(view.getAlpha(), equalTo(0f));
    }
}

File: uhabits-android/src/androidTest/java/org/isoron/uhabits/activities/habits/list/views/HintViewTest.kt
/*
 * Copyright (C) 2016-2021 Álinson Santos Xavier <git@axavier.org>
 *
 * This file is part of Loop Habit Tracker.
 *
 * Loop Habit Tracker is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by the
 * Free Software Foundation, either version 3 of the License, or (at your
 * option) any later version.
 *
 * Loop Habit Tracker is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.isoron.uhabits.activities.habits.list.views

import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.MediumTest
import com.nhaarman.mockitokotlin2.doReturn
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat
import org.isoron.uhabits.BaseViewTest
import org.isoron.uhabits.core.ui.screens.habits.list.HintList
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@MediumTest
class HintViewTest : BaseViewTest() {
    private lateinit var view: HintView
    private lateinit var list: HintList

    @Before
    override fun setUp() {
        super.setUp()
        list = mock()
        view = HintView(targetContext, list)
        measureView(view, 400f, 200f)
        val text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
        doReturn(true).whenever(list).shouldShow()
        doReturn(text).whenever(list).pop()
        view.showNext()
        skipAnimation(view)
    }

    @Test
    @Throws(Exception::class)
    fun testRender() {
        assertRenders(view, PATH + "render.png")
    }

    @Test
    @Throws(Exception::class)
    fun testClick() {
        assertThat(view.alpha, equalTo(1f))
        view.performClick()
        skipAnimation(view)
        assertThat(view.alpha, equalTo(0f))
    }

    companion object {
        const val PATH = "habits/list/HintView/"
    }
}
